zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -108,16 +111,24 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
 
         // TODO: Take gas cost of access list (EIP 2930) into consideration.
         // Use intrinsic gas
-        let intrinsic_gas_cost = select::expr(
+        let intrinsic_gas_cost_1 = select::expr(
             tx_is_create.expr(),
             GasCost::CREATION_TX.expr(),
             GasCost::TX.expr(),
         ) + tx_call_data_gas_cost.expr();
 
         // Check gas_left is sufficient
-        let gas_left = tx_gas.expr() - intrinsic_gas_cost;
+        let gas_left = tx_gas.expr() - intrinsic_gas_cost_1;
         let sufficient_gas_left = RangeCheckGadget::construct(cb, gas_left.clone());
 
+        let intrinsic_gas_cost_2 = select::expr(
+            tx_is_create.expr(),
+            GasCost::CREATION_TX.expr(),
+            GasCost::TX.expr(),
+        ) + tx_call_data_gas_cost.expr();
+        let is_gas_not_enough = intrinsic_gas_cost_2 - tx_gas.expr();
+        let is_sufficient_gas_left_invalid_tx = RangeCheckGadget::construct(cb, is_gas_not_enough.clone());
```

Comment:
Is this for checking that the tx gas limit is enough to pay for the intrinsic gas cost? If so I don't think we need to do anything for that here as explained in https://github.com/taikochain/zkevm-circuits/issues/25#issue-1411965257. Should be just the nonce and ETH balance where things can go wrong unless I'm mistaken.

Reply:
Correct. Updated to ETH balance.